### PR TITLE
MAINT: migrate stdin and random source to event model

### LINKED ISSUE #604: StdInSource to use Event model
**Is your feature request related to a problem? Please describe.**
Follow up on Event model change #538

**Describe the solution you'd like**

**Describe alternatives you've considered (Optional)**

**Additional context**

### LINKED ISSUE #605: RandomStringSource to use Event model
**Is your feature request related to a problem? Please describe.**
Follow up on Event model change #538

**Describe the solution you'd like**

**Describe alternatives you've considered (Optional)**

**Additional context**

### PR DESCRIPTION
**Pull request by chenqi0805**
### Description
This PR 

(1) migrate stdin and random source to use Event data model.
(2) also migrate TestBuffer to event model
 
### Issues Resolved
Resolves #604 
Resolves #605 
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by cmanning09 — CHANGES_REQUESTED** [review]
Hey @chenqi0805 , it looks like this PR includes a bunch of changes that are already in mainline. Can you clean up the PR and only include your changes to support stdin and random source plugins?

**Comment by chenqi0805** [comment]
> Hey @chenqi0805 , it looks like this PR includes a bunch of changes that are already in mainline. Can you clean up the PR and only include your changes to support stdin and random source plugins?

Yes. I put it on draft mode waiting for Taylor's PR to merge first. Now it is ready for review.

File: data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/source/RandomStringSource.java
Comment (dlvenable) [inline data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/source/RandomStringSource.java:37]: This came up in another PR. So this constant is now in three locations. 

I think it would be nice for the `Event` to have a builder which is creates a general single item Event.

Something like:

```
Event event = JacksonEvent.fromMessage(UUID.randomUUID().toString());
```

There might be a better name than `fromMessage`.

It could even set the `EventType`.

**Review by dlvenable — APPROVED** [review]
I'm fine to merge this, but I do want to clean up the `message` constant. We can fix it now, or as a follow-up PR.

File: data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/source/RandomStringSource.java
Comment (chenqi0805) [reply data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/source/RandomStringSource.java:37]: I like the idea. But since it touches the core I will reserve it for a separate PR. Opened an issue for that: #646

**Review by dlvenable — APPROVED** [review]

**Review by sshivanii — APPROVED** [review]

**Review by cmanning09 — APPROVED** [review]
I am good with these changes but you may want to watch how your are managing your changes locally. Looking at the commit messages for this PR I see other user's changes and other Merge's as well. This should be fine because you will have to squash before merging. You may find it simplier to manage and ensure your PR's are consistent by ensuring you rebase from main instead of merging. `git config pull.rebase true` should simplify this for you.

**Pull request merged**
